content/blog/real-time-smart-city-traffic-monitoring-using-microservices-based-streaming-architecture-part-2.md

```diff
@@ -0,0 +1,331 @@
+---
+title: Real-time Smart City Traffic Monitoring Using Microservices-based
+  Streaming Architecture (Part 2)
+date: 2021-10-12T02:21:16.357Z
+author: Mathieu Dumoulin
+authorimage: /img/Avatar1.svg
+---
+```json
+"authorDisplayName": "Mathieu Dumoulin",
+"category": "machine-learning",
+"publish": "2017-01-10T06:00:00.000Z"
+```
+
+**Editor’s Note:** MapR products and solutions sold prior to the acquisition of such assets by Hewlett Packard Enterprise Company in 2019 may have older product names and model numbers that differ from current solutions. For information about current offerings, which are now part of HPE Ezmeral Data Fabric, please visit <https://www.hpe.com/us/en/software/ezmeral-data-fabric.html>
+
+***Modern Open Source Complex Event Processing For IoT***
+
+This series of blog posts details my findings as I bring to production a fully modern take on Complex Event Processing, or CEP for short. In many applications, ranging from financials to retail and IoT applications, there is tremendous value in automating tasks that require to take action in real time. Putting aside the IT system and frameworks that would support this capability, this is clearly a useful capability.
+
+In the [first post of the series](https://developer.hpe.com/blog/better-complex-event-processing-at-scale-using-a-microservices-based-str/), I explain how CEP has evolved to meet this requirement and how the requirements for CEP can be met in a modern big data context. In short, I present an approach based on the best practices of modern architecture, microservices, and Kafka-style stream messaging, with an up-to-date open source business rule engine.
+
+In this second part, I’ll get more concrete and work through a working example using the system I propose. Let’s get started.
+
+## Smart City Traffic Monitoring
+
+We made a working demo for which the code will be released on the MapR GitHub. It is made to work on either the MapR Sandbox or using a real MapR cluster.
```

Comment:
Suggest you hotlink "the MapR Sandbox" and add the link to the blog post: https://developer.hpe.com/blog/getting-started-with-spark-on-mapr-sandbox/

Reply:
done